src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/AudienceValidationError.cs

```diff
@@ -10,27 +10,18 @@ namespace Microsoft.IdentityModel.Tokens
 {
     internal class AudienceValidationError : ValidationError
     {
-        private IList<string>? _tokenAudiences;
-        private IList<string>? _validAudiences;
-
-        // stack frames associated with AudienceValidationErrors
-        internal static StackFrame? ValidationParametersNull;
-        internal static StackFrame? AudiencesNull;
-        internal static StackFrame? AudiencesCountZero;
-        internal static StackFrame? ValidationParametersAudiencesCountZero;
-        internal static StackFrame? ValidateAudienceFailed;
-
         public AudienceValidationError(
             MessageDetail messageDetail,
-            ValidationFailureType failureType,
             Type exceptionType,
             StackFrame stackFrame,
             IList<string>? tokenAudiences,
-            IList<string>? validAudiences)
-            : base(messageDetail, failureType, exceptionType, stackFrame)
+            IList<string>? validAudiences,
+            ValidationFailureType? failureType = null,
```

Comment:
Not understanding why failureType should ever be null.

Reply:
`AudienceValidationError` has a default value of `AudienceValidationFailed` if null is passed.

You will notice that `ValidationError` does not allow to receive null values, but the subclasses that provide a default allow to do so.

Reply:
Removed the nullability.